### DISCUSSION TITLE
Apache Iceberg

### DISCUSSION THREAD

**Discussion by dforsber** [duckdb/duckdb#1668] [General]
Any plans with supporting Apache Iceberg?
https://iceberg.apache.org/

**Comment by hannes** [comment]
"We may have something in the works"

**Comment by mycaule** [comment]
What about Delta?

**Reply by Alex-Monahan** [reply]
One member of the DuckDB team is experimenting on their own, outside of work. Here is the repo, and the disclaimer in the Readme! Cool stuff as always, @Mause!
https://github.com/Mause/duckdb-deltatable-extension
"This is a purely experimental extension, not supported by the DuckDB Labs/Foundation, and is not suitable for production use."

**Reply by dennyglee** [reply]
Would love to help with this to make it officially supported it - please let me know how I can help, eh?! :)

**Reply by derekmahar** [reply]
What is Delta?

**Reply by Alex-Monahan** [reply]
Delta lake format!

**Comment by djouallah** [comment]
delta read works fine, there is a code example here using the package delta table
https://github.com/djouallah/duckdb_GCP_Storage/blob/main/app.py

**Reply by Alex-Monahan** [reply]
Yes, when combined with Apache Arrow! Thanks for the example of course!

**Comment by djouallah** [comment]
interesting twist, Apache iceberg is working on adding support to DuckDB
https://github.com/apache/iceberg/pull/6233

**Comment by mustafaakin** [comment]
Would be very interesting, looking forward to hear more about this and trying it.

**Reply by jakthom** [reply]
🤤

**Comment by Fokko** [comment]
For those that missed it: https://github.com/duckdb/duckdb_iceberg/ :)
